### DISCUSSION TITLE
Inline editing with `CopyFile` ?

### DISCUSSION THREAD

**Discussion by gardar** [CyberShadow/aconfmgr#163] [Q&A] [answered]
It seems that currently there are two ways to do in-line file editing. Either with `GetPackageOriginalFile` or `CreateFile`.

In a situation where I have a config I want to copy that differs between systems but that file does not come from a package, and echoing the whole content of the file with `CopyFile` doesn't feel right.

Ideally what I would like to do is something similar to the examples you provide with inline editing using `GetPackageOriginalFile`, but using a file I have under `files/` instead of getting it from a package.
Is there any way `CopyFile` or `CopyFileTo` can be used to achieve this?

**Comment by CyberShadow** [comment, marked as answer]
Yes, I don't think there's a helper for this. I would try something like this:

```bash
target="$(CreateFile /etc/passwd)"
cp "$config_dir"/templates/etc/passwd "$target"
sed -i 's/.../' "$target"
...
```

**Reply by gardar** [reply]
Perfect, that works!
